sn_protocol/src/messages/chunk_proof.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+/// The nonce provided by the verifier
+pub type Nonce = u64;
+
+/// The hash(record_value + nonce) that is used to prove the existence of a chunk
+#[derive(Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct ChunkProof([u8; 32]);
+
+impl ChunkProof {
+    pub fn new(record_value: &[u8], nonce: Nonce) -> Self {
```

Comment:
by making `ChunkProof` constructor taking a chunk_name instead of whole content will be enough,
and has the benefit of avoid un-necessary access to the whole content?
especialy at the previous place of 
```rust
                    // make sure the chunk is stored;
                    let chunk = Chunk::new(Bytes::from(std::fs::read(&chunk_path)?));
                    let res = client.verify_chunk_stored(&chunk).await;
```
which has to read from disk ?

Reply:
This can lead to dishonest nodes just storing the keys and discording the content.

Reply:
ah, right, make sense.